jinja2/base.html

```diff
@@ -140,12 +146,39 @@
               </ul>
             </div>
           </div>
-        </li>{% endif %}<li>{% include "includes/login.html" %}</li>
+      </li>{% endif %}<li class="nav-login">{% include "includes/login.html" %}</li>
       </ul>
     {% endif %}
     </div>
 
-    <nav id="main-nav" role="navigation"><ul><li><a href="{{ wiki_url('Web') }}">{{ _('Web Technologies') }}<i aria-hidden="true" class="icon-caret-down"></i></a>
+    <nav id="main-nav" class="nav-main" role="navigation"><ul>{% if waffle.flag('redesign') %}<li class="nav-main-item"><a href="{{ wiki_url('Web') }}">{{ _('Technologies') }}<i aria-hidden="true" class="icon-caret-down"></i></a>
+      <div class="submenu js-submenu" id="nav-tech-submenu">
+        <div class="submenu-column">
+          <ul>
+            <li><a href="{{ wiki_url('Web/HTML') }}">{{ _('HTML') }}</a></li>
+            <li><a href="{{ wiki_url('Web/CSS') }}">{{ _('CSS') }}</a></li>
+            <li><a href="{{ wiki_url('Web/JavaScript') }}">{{ _('JavaScript') }}</a></li>
+            <li><a href="{{ wiki_url('Web/Guide/Graphics') }}">{{ _('Graphics') }}</a></li>
+            <li><a href="{{ wiki_url('Web/HTTP') }}">{{ _('HTTP') }}</a></li>
+            <li><a href="{{ wiki_url('Web/API') }}">{{ _('APIs / DOM') }}</a></li>
+            <li><a href="{{ wiki_url('Mozilla/Add-ons/WebExtensions') }}">{{ _('WebExtensions') }}</a></li>
+            <li><a href="{{ wiki_url('Web/MathML') }}">{{ _('MathML') }}</a></li>
+          </ul>
+        </div></div><li class="nav-main-item"><a href="{{ wiki_url('Learn') }}">{{ _('References & Guides') }}<i aria-hidden="true" class="icon-caret-down"></i></a>
+        <div class="submenu js-submenu" id="nav-learn-submenu">
+        <div class="submenu-column">
+          <ul>
+            <li><a href="{{ wiki_url('Learn') }}">{{ _('Learn web development') }}</a></li>
+            <li><a href="{{ wiki_url('Web/Tutorials') }}">{{ _('Tutorials') }}</a></li>
+            <li><a href="{{ wiki_url('Web/Reference') }}">{{ _('References') }}</a></li>
+            <li><a href="{{ wiki_url('Web/Guide') }}">{{ _('Developer Guides') }}</a></li>
+            <li><a href="{{ wiki_url('Web/Accessibility') }}">{{ _('Accessibility') }}</a></li>
+            <li><a href="{{ wiki_url('Games') }}">{{ _('Game development') }}</a></li>
+            <li><a href="{{ wiki_url('Web') }}">{{ _('...more docs') }}</a></li>
+          </ul>
+        </div>
+      </div>
+      </li>{% else %}<li class="nav-main-item"><a href="{{ wiki_url('Web') }}">{{ _('Web Technologies') }}<i aria-hidden="true" class="icon-caret-down"></i></a>
```

Comment:
Bit: General formatting question. In the above I find lines 154, 167 and 181 hard to read because of the formatting. Is there a specific reason it is done this way?